pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1244,6 +1244,7 @@ func newChangeFrontierProcessor(
 	if cf.encoder, err = getEncoder(
 		ctx, encodingOpts, AllTargets(spec.Feed), spec.Feed.Select != "",
 		makeExternalConnectionProvider(ctx, flowCtx.Cfg.DB), sliMetrics,
+		enrichedSourceData{jobId: cf.spec.JobID.String()},
```

Comment:
nit: i'd prefer constructing an enrichedSourceProvider and passing it into getEncoder rather than passing in the sourceData and constructing it inside there. will make it easier to add it to avro as well.